RecoLocalTracker/SiStripClusterizer/plugins/ClustersFromZSRawProducer.cc

```diff
@@ -0,0 +1,300 @@
+/*
+ */
+#include "RecoLocalTracker/SiStripClusterizer/plugins/ClustersFromZSRawAlgorithm.h"
+
+#include "DataFormats/SiStripCluster/interface/SiStripCluster.h"
+#include "DataFormats/Common/interface/DetSetVectorNew.h"
+
+#include "DataFormats/FEDRawData/interface/FEDRawDataCollection.h"
+#include "EventFilter/SiStripRawToDigi/interface/SiStripFEDBuffer.h"
+#include "DataFormats/SiStripCommon/interface/SiStripConstants.h"
+
+#include "DataFormats/SiStripCluster/interface/SiStripClusterTools.h"
+#include "CondFormats/SiStripObjects/interface/FedChannelConnection.h"
+
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/Utilities/interface/Exception.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+#include <sstream>
+
+#include "FWCore/Utilities/interface/GCC11Compatibility.h"
+
+// #define VIDEBUG
+#ifdef VIDEBUG
+#include <iostream>
+#define COUT std::cout << "VI "
+#else
+#define COUT LogDebug("")
+#endif
+
+namespace {
+  std::unique_ptr<sistrip::FEDBuffer> fillBuffer(int fedId, const FEDRawDataCollection& rawColl) {
+    std::unique_ptr<sistrip::FEDBuffer> buffer;
+
+    // Retrieve FED raw data for given FED
+    const FEDRawData& rawData = rawColl.FEDData(fedId);
+
+    // Check on FEDRawData pointer
+    if
+      UNLIKELY(!rawData.data()) {
+        if (edm::isDebugEnabled()) {
+          edm::LogWarning("ClustersFromZSRawProducer") << " NULL pointer to FEDRawData for FED id " << fedId;
```

Comment:
ClustersFromRawProducer is using some standardized category names. 
it seems practical here to also use the same.
OTOH, if this logging is not essential for some downstream runtime analysis and just for debugging, perhaps it's better to change these two lines to one LogDebug

Reply:
whatever. It is historical